timemachine/fe/single_topology.py

```diff
@@ -1158,81 +904,83 @@ def assert_torsions_defined_over_non_linear_angles(system: GuestSystem | HostGue
-    assert len(dst_chiral_restr_idx_set.allowed_set.intersection(src_chiral_restr_idx_set.disallowed_set)) == 0
+    for i, j in core:
+        new_core_.append((i + num_host_atoms, j + num_host_atoms))
 
+    new_core = np.array(new_core_, dtype=np.int32)
 
-@dataclass
-class AlignedPotential:
-    idxs: NDArray[np.int32]
-    src_params: NDArray[np.float64]
-    dst_params: NDArray[np.float64]
-    mins: NDArray[np.float64]
-    maxes: NDArray[np.float64]
+    num_atoms_a = num_host_atoms + mol_a.GetNumAtoms()
+    num_atoms_b = num_host_atoms + mol_b.GetNumAtoms()
 
-    def interpolate(self, lamb):
-        raise NotImplementedError()
+    return AtomMapMixin(num_atoms_a, num_atoms_b, new_core)
 
 
-class AlignedBond(AlignedPotential):
-    def interpolate(self, lamb):
-        k_min = 0.1
-        params = batch_interpolate_harmonic_bond_params(
-            self.src_params, self.dst_params, lamb, k_min, self.mins, self.maxes
-        )
-        params = jnp.array(params).T
+def build_end_state_reference_all_pairs(
+    num_guest_atoms: int, host_nonbonded_all_pairs: BoundPotential[Nonbonded]
+) -> BoundPotential[Nonbonded]:
+    host_params = host_nonbonded_all_pairs.params
+    nb_params = np.concatenate([host_params, np.zeros((num_guest_atoms, host_params.shape[1]))])
 
-        return HarmonicBond(self.idxs).bind(params)
+    num_host_atoms = host_nonbonded_all_pairs.potential.num_atoms
+    host_atom_idxs = np.arange(num_host_atoms, dtype=np.int32)
 
+    host_exclusion_idxs = host_nonbonded_all_pairs.potential.exclusion_idxs
+    host_scale_factors = host_nonbonded_all_pairs.potential.scale_factors
+    host_cutoff = host_nonbonded_all_pairs.potential.cutoff
+    host_beta = host_nonbonded_all_pairs.potential.beta
+    total_num_atoms = num_host_atoms + num_guest_atoms
 
-class AlignedAngle(AlignedPotential):
-    def interpolate(self, lamb):
-        k_min = 0.05
-        params = batch_interpolate_harmonic_angle_params(
-            self.src_params, self.dst_params, lamb, k_min, self.mins, self.maxes
-        )
-        params = jnp.array(params).T
-        return HarmonicAngle(self.idxs).bind(params)
+    nonbonded_all_pairs = Nonbonded(
+        total_num_atoms, host_exclusion_idxs, host_scale_factors, host_beta, host_cutoff, host_atom_idxs
+    )
+    return nonbonded_all_pairs.bind(nb_params)
 
 
-class AlignedTorsion(AlignedPotential):
-    def interpolate(self, lamb):
-        params = batch_interpolate_periodic_torsion_params(
-            self.src_params, self.dst_params, lamb, self.mins, self.maxes
-        )
-        params = jnp.array(params).T
-        return PeriodicTorsion(self.idxs).bind(params)
+def build_end_state_reference_ixn_group(
+    mol: Chem.Mol, nonbonded_all_pairs: BoundPotential[Nonbonded], ff
+) -> BoundPotential[NonbondedInteractionGroup]:
+    guest_q = ff.q_handle.parameterize(mol)
+    guest_lj = ff.lj_handle.parameterize(mol)
+    guest_w = jnp.zeros_like(guest_q)
```

Comment:
A bit annoying, this is required for 
`test_single_topology_combined.py::test_nonbonded_host_params_independent_of_lambda()` to be traceable through `jax.jit`

From what I can tell, the only reason `guest_w` needs to be `jnp` since `guest_q` is emitted because our ff handlers emit `jnp`, even though it could technically be a numpy array.